util/network/src/error.rs

```diff
@@ -84,11 +84,16 @@ error_chain! {
 	foreign_links {
 		SocketIo(IoError) #[doc = "Socket IO error."];
 		Io(io::Error) #[doc = "Error concerning the Rust standard library's IO subsystem."];
-		AddressParse(net::AddrParseError) #[doc = "Error concerning the network address parsing subsystem."];
 		Decompression(snappy::InvalidInput) #[doc = "Decompression error."];
 	}
 
 	errors {
+		#[doc = "Error concerning the network address parsing subsystem."]
+		AddressParse(err: Option<io::Error>) {
```

Comment:
I'm not sure if it's worth having `Option<io::Error>` here. Maybe it should be just `AddressParse()` ? `io:Error` is always the same https://doc.rust-lang.org/src/std/net/addr.rs.html#891